app/controllers/app/main_controller.rb

```diff
@@ -1,5 +1,15 @@
 class App::MainController < App::BaseController
     def home
-        
+      start_time = 1.week.ago
+      end_time = Time.now
+    
+      foods = Food.where(consumed_at: start_time..end_time)
+      trainings = Training.where(exercise_time: start_time..end_time)
+    
+      @total_meals = foods.size
+      @total_calories = foods.sum(:calories)
+      @number_of_exercises_performed = trainings.size
+      @number_of_reps_performed = trainings.sum(:number_of_reps)
+      @number_of_sets_performed = trainings.sum(:number_of_sets)
```

Comment:
As a good practice, we avoid having many instance variables in a controller action.
You can create an object that represents your view logic and expose that object as an instance variable instead.

By applying this technique you would have this controller action like this: 

```ruby
class App::MainController < App::BaseController
  def home
    @dashboard = Dashboard.new(current_user)
  end
end    
```

In the view you could call methods on that object like:

```erb
<p>Meals eaten: <%= @dashboard.total_meals %></p>
<p>Calories eaten: <%= @dashboard.total_calories %></p>
```

By extracting all those items to an object you are able to test its logic very well without depending on the controller or view context.

You can also isolate the logic used in the view for the charts.